### PR TITLE
Update DG with new model

### LINKED ISSUE #150: Update DG with new model

### PR DESCRIPTION
**Pull request by njxue**
- Changed the UML diagram for Model component to show only the higher-level classes
- Added object diagram for TA-Assist to describe UniqueList
- Added class diagrams to describe lower-level classes in Model
- Added some new descriptions about the Model (@RezwanArefin01 might want to add more)
- Edited sequence diagram for focus command

Misc
- Enabled Assertions in build.gradle as required (under one of the tasks in Week 10)

#150

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -121,13 +121,21 @@ How the parsing works:
...
-  * all `ModuleClass` objects are contained in a `UniqueModuleClassList` object.
+  * all `Student` objects are contained in a `UniqueList` object.
+  * all `ModuleClass` objects are also contained in a `UniqueList` object.
 * stores the currently 'selected' `Student` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Student>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
 * stores the currently 'focused' `ModuleClass` object.
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
 * does not depend on any of the other three components (i.e. `Ui`, `Logic` and `Storage`) as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components.
 
+The `UniqueList` class is a generic class that stores a collection of unique elements. In TA Assist, a `UniqueList` stores either all the `Student` objects or all the `ModuleClass` objects.
```
Comment (Berted) [inline docs/DeveloperGuide.md:133]: Perhaps should mention how the equivalence of two elements is checked by the functional interface `Identity<T>`?

**Review by Berted — CHANGES_REQUESTED** [review]
Nice work! However, a section can be made more complete. In addition, would it be possible to enlarge `FocusCommandSequenceDiagram.png`? Right now it's quite blurry.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (njxue) [reply docs/DeveloperGuide.md:133]: 👌 added this in the next commit though I think Rezwan might consider elaborating further.
I can't really judge the image quality of the sequence diagram from intellij (it's not very accurate) so let me know if it's still blurry!

Comment (njxue) [reply docs/DeveloperGuide.md:133]: 👌 added this in the next commit though I think Rezwan might consider elaborating further.
I can't really judge the image quality of the sequence diagram from intellij (it's not very accurate) so let me know if it's still blurry!

**Review by Berted — APPROVED** [review]
LGTM!

**Pull request merged**